tools/file_replace.zig

```diff
@@ -0,0 +1,124 @@
+// NOTE(blackedout): Modified from https://ziglang.org/learn/build-system/#project-tools (2025-10-29)
+// This program should not be used for many occurrences of patterns (with short length) as each pattern invokes two file writes.
+
+const std = @import("std");
+
+const usage =
+	\\Usage: ./file_replace pattern replacement file
+;
+
+pub fn main() !void {
+	var arena_state = std.heap.ArenaAllocator.init(std.heap.page_allocator);
+	defer arena_state.deinit();
+	const arena = arena_state.allocator();
+
+	const args = try std.process.argsAlloc(arena);
+
+	var optPattern: ?[]const u8 = null;
+	var optReplacement: ?[]const u8 = null;
+	var optFilePath: ?[]const u8 = null;
+
+	{
+		var i: usize = 1;
+		while(i < args.len) : (i += 1) {
+			const arg = args[i];
+			if(std.mem.eql(u8, "-h", arg) or std.mem.eql(u8, "--help", arg)) {
+				try std.fs.File.stdout().writeAll(usage);
+				return std.process.cleanExit();
+			} else if(i == 1) {
+				optPattern = args[i];
+			} else if(i == 2) {
+				optReplacement = args[i];
+			} else if(i == 3) {
+				optFilePath = args[i];
+			} else {
+				fatal("superfluous arg: '{s}'", .{arg});
+			}
+		}
+	}
+
+	const pattern = optPattern orelse fatal("missing pattern", .{});
+	if(pattern.len == 0) {
+		fatal("pattern is empty (not allowed)", .{});
+	}
+	const replacement = optReplacement orelse fatal("missing replacement", .{});
+	const filePath = optFilePath orelse fatal("missing file", .{});
+
+	const outputFilePathSuffix = ".tmp";
+	const tmpOutputFilePathLength = filePath.len + outputFilePathSuffix.len;
+	var outputFilePath = arena.alloc(u8, tmpOutputFilePathLength) catch fatal("path allocation failed", .{});
+	defer arena.free(outputFilePath);
+	std.mem.copyForwards(u8, outputFilePath[0..filePath.len], filePath);
+	std.mem.copyForwards(u8, outputFilePath[filePath.len..], outputFilePathSuffix);
+
+	{
+		var inputFile = std.fs.cwd().openFile(filePath, .{}) catch |err| {
```

Comment:
Instead of openFile and reading manually, I'd suggest `std.fs.cwd().readFileAlloc` to do it all in one operation.